geoana/em/base.py

```diff
@@ -7,99 +7,111 @@
 import properties
 from scipy.constants import mu_0, pi, epsilon_0
 
+from .. import spatial
+
+
+###############################################################################
+#                                                                             #
+#                              Base Classes                                   #
+#                                                                             #
+###############################################################################
+
 
 class BaseEM(properties.HasProperties):
+    """
+    Base class for electromanetics. Contains physical properties that are
+    relevant to all problems that use Maxwell's equations
+    """
 
     mu = properties.Float(
-        help='Magnetic permeability.',
+        "Magnetic permeability (H/m)",
         default=mu_0,
         min=0.0
     )
 
     sigma = properties.Float(
-        help='Electrical conductivity (S/m)',
+        "Electrical conductivity (S/m)",
         default=1.0,
         min=0.0
     )
 
     epsilon = properties.Float(
-        help='Permitivity value',
+        "Permitivity value (F/m)",
         default=epsilon_0,
         min=0.0
     )
 
 
 class BaseDipole(BaseEM):
+    """
+    Base class for dipoles.
+    """
 
     orientation = properties.Vector3(
-        help='orientation of dipole',
-        default='X',
+        "orientation of dipole",
+        default="X",
         length=1.0
     )
 
     location = properties.Vector3(
-        help='location of the electric dipole source',
-        default='ZERO'
-    )
-
-    def offset_from_location(self, xyz):
-
-        # TODO: validate stuff
-        # xyz = Utils.asArray_N_x_Dim(xyz, 3)
-
-        return np.c_[
-            xyz[:, 0] - self.location[0],
-            xyz[:, 1] - self.location[1],
-            xyz[:, 2] - self.location[2]
-        ]
-
-    def distance_from_location(self, xyz):
-        return np.sqrt((self.offset_from_location(xyz)**2).sum(axis=1))
-
-
-class BaseFDEM(BaseEM):
-
-    frequency = properties.Float(
-        help='Source frequency (Hz)',
-        default=1e2,
-        min=0.0
+        "location of the electric dipole source",
+        default="ZERO"
     )
 
-    @property
-    def omega(self):
-        return 2.0*pi*self.frequency
-
-    @property
-    def sigma_hat(self):
-        return self.sigma + 1j*self.omega*self.epsilon
-
-    @property
-    def wave_number(self):
-        np.sqrt(
-            self.omega**2. * self.mu * self.epsilon -
-            1j * self.omega * self.mu * self.sigma
+    def vector_distance(self, xyz):
+        """
+        Vector distance from the dipole location
+        :param numpy.ndarray xyz: grid
+        """
+        return spatial.vector_distance(xyz, self.location)
```

Comment:
Is there any point in having these defined on the class if it is just a pass through call? Maybe - just curious your thinking here.

Reply:
The dipole has a location and the distance from it is important for solving for the fields, so it slightly cleans up the code to say `self.vector_distance(xyz)` in the computation of the fields. I am also thinking a bit along the lines of interacting with a dipole: you should be able to compute the distance from it since it has a location, so just asking the question: "what can I do with a dipole", computing the vector distance from it seemed like it might be handy. Not a strong argument, but a couple thoughts. What do you think?